### PR TITLE
Implement basic completion

### LINKED ISSUE #6: Implement basic completion
https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#completion-request

### PR DESCRIPTION
**Pull request by natgabb**
Closes #6

### CODE REVIEW

File: groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovyTextDocumentService.java [hunk 1]
```diff
@@ -185,4 +190,63 @@ private void publishDiagnostics(Set<Diagnostic> diagnostics) {
         publishDiagnostics.accept(paramsBuilder.build());
     }
 
+    private static CompletionList createCompletionListFromSymbols(Set<SymbolInformation> symbols) {
+        CompletionListBuilder builder = new CompletionListBuilder();
+        builder.incomplete(false);
+        if (symbols == null) {
+            return builder.build();
```
Comment (jiahuijiang) [inline groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovyTextDocumentService.java]: looks like the `items` field will be undefined if we don't set anything in the builder.
You can consider doing something like:

```
    if (symbols == null) {
        return new CompletionListImpl(false, Lists.newArrayList());
    }
    CompletionListBuilder builder = new CompletionListBuilder();
    builder.incomplete(false);
    symbols.forEach(symbol -> {
        builder.item(new CompletionItemBuilder()
                .label(symbol.getName())
                .kind(symbolKindToCompletionItemKind(symbol.getKind()))
                .build());
    });
    return builder.build();
```

Comment (natgabb) [inline groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovyTextDocumentService.java]: Fixed! :) and I added a test that checks this

File: groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovyTextDocumentService.java [hunk 2]
```diff
@@ -185,4 +190,63 @@ private void publishDiagnostics(Set<Diagnostic> diagnostics) {
...
+                    .kind(symbolKindToCompletionItemKind(symbol.getKind()))
+                    .build());
+        });
+        return builder.build();
+    }
+
+    @SuppressWarnings("checkstyle:cyclomaticcomplexity") // this is not complex behaviour
+    private static CompletionItemKind symbolKindToCompletionItemKind(SymbolKind kind) {
+        switch (kind) {
+            case Array:
```
Comment (gracew) [inline groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovyTextDocumentService.java]: I think it's easier to read if you use fall-through:

```
switch (kind) {
    case Array:
    case Boolean:
        return CompletionItemKind.Value;
    ...
}
```

Comment (natgabb) [inline groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovyTextDocumentService.java]: done

File: groovy-language-server/src/test/java/com/palantir/ls/groovy/GroovyTextDocumentServiceTest.java [hunk 3]
```diff
@@ -203,12 +217,50 @@ public void testDocumentSymbols_relativePath() throws InterruptedException, Exec
     @Test
     public void testReferences() throws InterruptedException, ExecutionException {
         // HACK, blocked on https://github.com/TypeFox/ls-api/issues/39
-        ReferenceParams params =
-                (ReferenceParams) new ReferenceParamsBuilder().context(false).position(5, 5).textDocument("uri")
-                        .uri("uri").build();
+        ReferenceParams params = (ReferenceParams) new ReferenceParamsBuilder()
+                .context(false).position(5, 5)
```
Comment (gracew) [inline groovy-language-server/src/test/java/com/palantir/ls/groovy/GroovyTextDocumentServiceTest.java]: (nit) put position on its own line

**Comment by gracew** [comment]
+1

**Pull request merged**